app/src/main/java/com/randomappsinc/studentpicker/database/DataSource.java

```diff
@@ -134,6 +136,40 @@ public void renameList(int listId, String newName) {
         close();
     }
 
+    public Set<ListDO> getAllNamesSet() {
+        Set<ListDO> allNamesSet = new HashSet<>();
```

Comment:
Hmm, I feel like this is sort of hard to read? If I were to do this, I would do a HashMap of list ID -> list. For every row returned by the query (name + list information), I would grab the list. If it's null, add to its list of names. Otherwise, instantiate a new list, attach the name to it, and put it in the map. At the end, you can return a List of ListDOs instead of a Set.